### DISCUSSION TITLE
getBalance returns 0

### DISCUSSION THREAD

**Discussion by Rubon72** [ethereum/web3.js#6943] [Q&A]
I run in console
`
const web3 = new Web3('https://eth-mainnet.g.alchemy.com/v2/'+ALCHEMY_API_KEY);
await web3.eth.getBalance("0x"+PUBLIC_WALLET_ADDRESS)
`
it's return 0n
But on my ballance 0.6  ETH

**Comment by TatyOko28** [comment]
If web3.eth.getBalance() returns 0n while the wallet has ETH, possible causes include:
The wallet address might be incorrect (ensure it includes 0x).
 The Ethereum network or RPC provider might not be correctly connected.
 The API key may have rate limits or restrictions.
The wallet may be on a different network (e.g., Sepolia, Goerli, or another chain) instead of Ethereum Mainnet.
